lib/pure/random.nim

```diff
@@ -617,15 +622,60 @@ proc shuffle*[T](x: var openArray[T]) =
 
   shuffle(state, x)
 
-when not defined(nimscript) and not defined(standalone):
-  import times
+when not defined(standalone):
+  when defined(nimscript):
+    import std/hashes
+
+    var baseState = block:
+      var ret = Rand(
+        a0: CompileTime.hash.Ui,
+        a1: CompileDate.hash.Ui)
+      if not ret.isValid:
+        ret = DefaultRandSeed
+      ret
+  elif defined(js):
+    import std/times
+  else:
+    import std/[hashes, os, sysrand, monotimes]
+
+    var baseState: Rand = block:
```

Comment:
this is problematic, `import rand` shouldn't trigger code that executes at the top-level; it should be lazily initialized on 1st use

Reply:
Thank you! I fixed it.
`baseState` is initialized in initRand() proc.